LEAF_Request_Portal/templates/reports/LEAF_import_data.tpl

```diff
@@ -561,183 +638,168 @@
                             });
 
                             function selectRowToAnswer(i) {
-                                return new Promise(function(resolve, reject){
-                                    var titleIndex = i;
-                                var completed = 0;
-                                var row = sheet_data.cells[titleIndex];
-                                var requestData = new Object();
-                                function answerQuestions() {
-                                    return new Promise(function(resolve, reject){
-                                    if (completed >= indicatorArray.length) {
-                                        requestData['title'] = titleInputNew.val() + '_' + titleIndex;
-                                        makeRequests(newCategoryID, requestData).then(function(){
-                                            resolve();
-                                        });
-                                    } else if (titleIndex <= sheet_data.cells.length - 1) {
-                                        var currentCol = newFormIndicators.find('tbody > tr:eq(' + completed + ') > td:first').html();
-                                        var currentFormat = newFormIndicators.find('tbody > tr:eq(' + completed + ') > td:eq(2) > select > option:selected').val();
-                                        switch (currentFormat) {
-                                            case 'orgchart_employee':
-                                                var sheetEmp = typeof (row[currentCol]) !== "undefined" && row[currentCol] !== null && !$.isNumeric(row[currentCol]) ? row[currentCol].toString() : '';
-                                                nexusAPI.Employee.getByEmailNational({
-                                                    'onSuccess': function (user) {
-                                                        var res = Object.keys(user);
-                                                        var emp = user[res[0]];
-                                                        if (typeof (emp) !== "undefined" && emp !== null && res.length === 1) {
-                                                            nexusAPI.Employee.importFromNational({
-                                                                'onSuccess': function (results) {
-                                                                    if (!isNaN(results)) {
-                                                                        requestData[indicatorArray[completed]] = parseInt(results);
-                                                                    } else {
-                                                                        requestData['failed'] = currentCol + titleIndex + ': Employee ' + sheetEmp + ' not found. Error: '+ results;
-                                                                    }
+                                return new Promise(function(resolve, reject) {

```

Comment:
Convince me to allow this.  A return that returns promise that returns a promise and/or does a hundred lines of work, all inside a return.

Reply:
I had mentioned this to Michael Gao too.  The page as a whole can use a more extensive refactor.  He wants to fix the major issues now and then address further refactoring in a future story.  But yes, it's functions that return promises that have function declarations in them that return promises that have function declarations that return more promises.

I pulled some methods out, but untangling everything would require a more significant amount of time.